### DISCUSSION TITLE
Querying VARCHAR column using PHP integer value in the Query Builder

### DISCUSSION THREAD

**Discussion by pratheepbg-dckap** [laravel/framework#38712] [Q&A]
- Laravel Version: 8.55.0
- PHP Version: 7.4.23
- Database Driver & Version: mysql (10.4.21-MariaDB)

### Description:

Greetings Artisans !!!

When Querying a database table on a `VARCHAR` column  with an `integer` value in the **Query Builder**, the SQL query is created like `column =  1` while querying the Database.  

```sql
SELECT * FROM `table_name` WHERE `varchar_column` = 1 LIMIT 1;
```
  
This is resulting in the records that starts with the _same characters in that number_

```php
DB::table('table_name')->where('varchar_column', '=', 1527)->get();
```

we are getting results with the values like `1527`, `1527-1` & so on. This causes problem when mixed with `->orderBy()` and getting single record using `->first()` and messes with the Computation.  
  
This can be solved by passing `string` type in the `where` method like `->where('varchar_column', '=', (string) 1527)`.
Is there any possible ways for Laravel to determine **Column type** and performing the casting automatically?

### Steps To Reproduce:

1. Create a database table with **VARCHAR** Column in the _mysql_ database. Insert values starting with the numbers like `1527`, `1527-1`, `1527-a1` & so on for the Varchar column.
2. Query the table using Laravel Query Builder and placing an `where` constraint providing _PHP integer_ value
3. The Result contains all the values starting with that numbers

**Comment by Muthukumar-Kannan** [comment]
I also facing this issue a long time back.
